### PR TITLE
>#208 stm32: UART listen mode

### PR DESCRIPTION
**Pull request by forGGe**
Includes listen mode tests, improved error handling in USART bus and
corrected strings in the suite generator.

### CODE REVIEW

File: platform/stm32/export/aux/usart_bus.hpp [hunk 1]
```diff
@@ -307,21 +327,17 @@ void usart_bus<dev>::set_rx(uint8_t *rx, size_t size)
 template<usart_device dev>
 void usart_bus<dev>::set_tx(size_t size, uint8_t fill_byte)
 {
-    if (!inited()) {
-        return;
-    }
+    // TODO: implement
+    ecl_assert_msg(0, "Not implemeted!");
```
Comment (RostakaGmfun) [inline platform/stm32/export/aux/usart_bus.hpp]: How about using `static_assert` here?

Comment (forGGe) [inline platform/stm32/export/aux/usart_bus.hpp]: Static assert will hit, even if there is no function call immediately after template instance is created.

Comment (forGGe) [inline platform/stm32/export/aux/usart_bus.hpp]: Even worse - it will be there even [if template is not touched](https://godbolt.org/g/EUqTKV)

Comment (RostakaGmfun) [inline platform/stm32/export/aux/usart_bus.hpp]: Oh, didn't know

Comment (forGGe) [inline platform/stm32/export/aux/usart_bus.hpp]: No problem! :)

**Pull request merged**